File: examples/agent-framework-webhooks/typescript/src/adapters/fake.ts
import type { InboundEmail } from "@e2a/sdk/v1";

import type { ReplyAgent } from "../contracts.js";
import { emailPrompt } from "../prompt.js";

export class FakeReplyAgent implements ReplyAgent {
  readonly prompts: string[] = [];

  constructor(readonly response = "Fake") {}

  get callCount(): number {
    return this.prompts.length;
  }

  async reply(email: InboundEmail, _conversationId: string): Promise<string> {
    this.prompts.push(emailPrompt(email));
    return this.response;
  }
}

File: examples/agent-framework-webhooks/python/agent_webhooks/adapters/fake.py
"""Deterministic no-key reply adapter for tests and dry runs."""

from e2a import AsyncInboundEmail

from agent_webhooks.prompt import email_prompt


class FakeReplyAgent:
    def __init__(self, response: str = "Fake") -> None:
        self.response = response
        self.prompts: list[str] = []

    @property
    def call_count(self) -> int:
        return len(self.prompts)

    async def reply(
        self, email: AsyncInboundEmail, conversation_id: str
    ) -> str:
        del conversation_id
        self.prompts.append(email_prompt(email))
        return self.response
